### ISSUE TITLE
Buffer local textobj mappings in documentation

### ISSUE THREAD

**Issue by lukelbd** [kana/vim-textobj-user#72]
This plugin is great for defining *filetype specific* text objects, but it seems they are often [defined globally](https://github.com/rbonvall/vim-textobj-latex/blob/master/ftplugin/tex/textobj-latex.vim) because users don't know how to make them specific to a buffer.

The ability to add `<option>` flags to mappings, including the `<buffer>` flag, should be stated more clearly in the documentation. I thought this was impossible for a long time before looking at the source code.

Example: The below is in my `.vimrc` and defines a bunch of latex related text objects.

```vim
" Declare text objects
augroup tex_textobjs
  au!
  au FileType tex call textobj#user#plugin('latex', s:tex_textobjs_dict)
augroup END

" TeX plugin definitions
" Copied from: https://github.com/rbonvall/vim-textobj-latex/blob/master/ftplugin/tex/textobj-latex.vim
let s:tex_textobjs_dict = {
  \   'environment': {
  \     'pattern': ['\\begin{[^}]\+}.*\n', '\\end{[^}]\+}.*$'],
  \     'select-a': '<buffer> aT',
  \     'select-i': '<buffer> iT',
  \   },
  \  'command': {
  \     'pattern': ['\\\S\+{', '}'],
  \     'select-a': '<buffer> at',
  \     'select-i': '<buffer> it',
  \   },
  \  'paren-math': {
  \     'pattern': ['\\left(', '\\right)'],
  \     'select-a': '<buffer> a(',
  \     'select-i': '<buffer> i(',
  \   },
  \  'bracket-math': {
  \     'pattern': ['\\left\[', '\\right\]'],
  \     'select-a': '<buffer> a[',
  \     'select-i': '<buffer> i[',
  \   },
  \  'curly-math': {
  \     'pattern': ['\\left\\{', '\\right\\}'],
  \     'select-a': '<buffer> a{',
  \     'select-i': '<buffer> i{',
  \   },
  \  'angle-math': {
  \     'pattern': ['\\left<', '\\right>'],
  \     'select-a': '<buffer> a<',
  \     'select-i': '<buffer> i<',
  \   },
  \  'dollar-math-a': {
  \     'pattern': '[$][^$]*[$]',
  \     'select': '<buffer> a$',
  \   },
  \  'dollar-math-i': {
  \     'pattern': '[$]\zs[^$]*\ze[$]',
  \     'select': '<buffer> i$',
  \   },
  \  'quote': {
  \     'pattern': ['`', "'"],
  \     'select-a': "<buffer> a'",
  \     'select-i': "<buffer> i'",
  \   },
  \  'quote-double': {
  \     'pattern': ['``', "''"],
  \     'select-a': '<buffer> a"',
  \     'select-i': '<buffer> i"',
  \   },
  \ }
```

**Comment by kana** [comment]
Thank you for the report.  I forgot to write about `<buffer>` in the document.  I'll write it later.

**Comment by kana** [comment]
I wrote [an example to define text objects for a specific filetype](https://github.com/kana/vim-textobj-user/pull/73/files#diff-04c6e90faac2675aa89e2176d2eec7d8R95-R111).  It is based on your example, but I modified a bit.

Your example works fine, but it redefines also all of internal stuffs for the text objects whenever tex file is opened.  It would be better to only define buffer-local key mappings on `FileType` event if possible.
